packages/flutter/lib/src/material/predictive_back_page_transitions_builder.dart

```diff
@@ -0,0 +1,299 @@
+  final ValueGetter<bool> getIsCurrent;
+  final Widget child;
+
+  Widget _secondaryAnimatedBuilder(BuildContext context, Widget? child) {
+    final Size size = MediaQuery.sizeOf(context);
+    final double screenWidth = size.width;
+    final double xShift = (screenWidth / 20) - 8;
+
+    final bool isCurrent = getIsCurrent();
+    final Tween<double> xShiftTween = isCurrent
+        ? ConstantTween<double>(0)
+        : Tween<double>(begin: xShift, end: 0);
+    final Animatable<double> scaleTween = isCurrent
+        ? ConstantTween<double>(1)
+        : TweenSequence<double>(<TweenSequenceItem<double>>[
+            TweenSequenceItem<double>(
+                tween: Tween<double>(begin: 0.95, end: 1), weight: 65.0),
+            TweenSequenceItem<double>(
+                tween: Tween<double>(begin: 1, end: 1), weight: 35.0),
+          ]);
+    final Animatable<double> fadeTween = isCurrent
+        ? ConstantTween<double>(1)
+        : TweenSequence<double>(<TweenSequenceItem<double>>[
+            TweenSequenceItem<double>(
+                tween: Tween<double>(begin: 1.0, end: 0.8), weight: 65.0),
+            TweenSequenceItem<double>(
+                tween: Tween<double>(begin: 1, end: 1), weight: 35.0),
+          ]);
+
+    return Transform.translate(
+      offset: Offset(xShiftTween.animate(secondaryAnimation).value, 0),
+      child: Transform.scale(
+        scale: scaleTween.animate(secondaryAnimation).value,
+        child: Opacity(
+          opacity: fadeTween.animate(secondaryAnimation).value,
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  Widget _primaryAnimatedBuilder(BuildContext context, Widget? child) {
+    final Size size = MediaQuery.sizeOf(context);
+    final double screenWidth = size.width;
+    final double xShift = (screenWidth / 20) - 8;
+
+    final Animatable<double> xShiftTween = TweenSequence<double>(<TweenSequenceItem<double>>[
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: 0.0, end: 0.0),
+        weight: 65.0,
+      ),
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: xShift, end: 0.0),
+        weight: 35.0,
+      ),
+    ]);
+    final Animatable<double> scaleTween = TweenSequence<double>(<TweenSequenceItem<double>>[
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: 1.0, end: 1.0),
+        weight: 65.0,
+      ),
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: 0.95, end: 1.0),
+        weight: 35.0,
+      ),
+    ]);
+    final Animatable<double> fadeTween = TweenSequence<double>(<TweenSequenceItem<double>>[
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: 0.0, end: 0.0),
+        weight: 65.0,
+      ),
+      TweenSequenceItem<double>(
+        tween: Tween<double>(begin: 0.95, end: 1.0),
+        weight: 35.0,
+      ),
+    ]);
+
+    return Transform.translate(
+      offset: Offset(xShiftTween.animate(animation).value, 0),
```

Comment:
nit: prefer double literals for double constants from [style-guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-double-literals-for-double-constants), `0` -> `0.0`.